### ISSUE TITLE
[documentation] Add reasoning for rules in documentation pages

### ISSUE THREAD

**Issue by Samuel-Therrien-Beslogic** [angular-eslint/angular-eslint#1447]
**Description and reproduction of the issue**

Rules documentation should help devs understand why it exists, what issue/codesmell/bad practice it's trying to solve, or if it's purely stylistic.
See for example https://github.com/angular-eslint/angular-eslint/blob/main/packages/eslint-plugin-template/docs/rules/button-has-type.md . Why should I specify type for regular buttons? What can happen if I don't? Help me, as a developer and user of this plugin, understand if this rule fits my needs.

**Versions**

N/A

- [ ] I have tried restarting my IDE and the issue persists.
- [ ] I have updated to the latest supported version of the packages and checked my `ng version` output per the instructions given here.

**Comment by P4** [comment]
The original codelyzer docs provide a rationale for most of the rules (example: http://codelyzer.com/rules/template-no-call-expression/), no idea why it was skipped when the rules got ported over.

I find myself going back there occasionally to explain to other people why a given rule exists, but it feels weird to reference docs for a completely different tool that's no longer maintained.

**Comment by JamesHenry** [comment]
I agree things can be more fleshed out in this regard. It's important to note that the documentation is auto-generated from the implementations and unit tests. This is hugely beneficial because it ensures that it is always up to date and exactly correct in terms of behavior.

I will happily accept any enhancements to the documentation generation tooling to be able to pass through more in depth documentation for each rule, as well as subsequent PRs to build out more in depth documentation coverage of rules themselves.

**Comment by json-derulo** [comment]
@JamesHenry for the rules which already have a rationale in the [codelyzer docs](http://codelyzer.com/rules/), do you also prefer to one PR per rule, or would it be an option here to create one big PR for migrating all the codelyzer rationales?

**Comment by JamesHenry** [comment]
I'm pleased to say that all rules now have a detailed rationale for why they exist, and all new rules going forward are required to have one otherwise they will fail a new local lint rule I set up
